addon/components/pix-checkbox.hbs

```diff
@@ -0,0 +1,12 @@
+<div class="pix-checkbox">
+  <label class="pix-checkbox__label {{this.labelSizeClass}}" for={{this.id}}>
+    {{this.label}}
+  </label>
+  <input
+    id={{this.id}}
+    type="checkbox"
+    aria-label={{this.ariaLabel}}
+    class="pix-checkbox__input {{if @isIndeterminate ' pix-checkbox__input--indeterminate'}}"
+    ...attributes
```

Comment:
[question] C'est une question que je me pose sur pas mal de composant qu'on a.

Si l'on doit surcharger le css de pix-checkbox pour du positionnement dans une application ? 

- Ne vaudrait-il pas mieux poser le ...attributes sur l'element parent du composant ? ( le `pix-checkbox`)
- Cibler la class pix-checkbox dans l'application (breaking change si le nom du composant change)
- Ajouter un attribut `@class` sur l'element parent pour pouvoir potentiellement faire la surcharge facielement ?

Reply:
Je suis d'accord, nos composants ont souvent cette problématique là et dans le code je vois soit de la surcharge soit le composant englobé dans une div pour pouvoir la positionner 🤔

Reply:
Le problème de remonter le `...attributes` c'est qu'on perds la possibilité d'ajouter d'autres attributs "fonctionnels" à l'élélement important (ici l'`input` et non la `div` parent). Ex: si on veut rajouter un `role=button` (ou autre) on le veut sur l'input.

Si on doit surcharger le css de la pix-checkbox c'est qu'elle n'est pas encore bien faite à mon sens. Le but du design system c'est de ne pas à avoir à la modifier. Si la problématique est le positionnement , une div englobant la PixCheckbox pourrait faire l'affaire 🤔 

Est-ce que tu as un exemple concret dans lequel tu as besoin de surcharger la PixCheckbox en selectionnant le parent @xav-car  ?

Reply:
De mon point de vue le `...attributes` devrait être utilisé pour rajouter des attributs fonctionnels au composant et non à une surcharge CSS

Reply:
Le composant à l'heure actuel en flex, prendra tout l'espace disponible quoi qu'il arrive, ce qui n'est pas forcément quelque chose que l'on souhaite partout.
Pour du placement inline / block, une checkbox dans un bandeau filtre est afficher différement que dans un formulaire. C'est un cas qui nous a causé du souci dans les bandeaux filtre pour des composants en flex (multi select ou le select il me semble). 

Personnelement, rajouter un nouvel element html pour contenir un effet de bord de composant ne me semble pas une bonne approche. Sur une dimension de performance rendering / painting.

Reply:
> Le composant à l'heure actuel en flex, prendra tout l'espace disponible quoi qu'il arrive, ce qui n'est pas forcément quelque chose que l'on souhaite partout.

Les seules checkbox que je connais sur Pix App sont celle dans le formulaire d'inscription et celles dans les épreuves et dans les deux  cas on veut qu'elles prennent toute la largeur du parent j'ai l'impression. 
Du coup j'ai du mal à voir dans quels autres cas on ne veut pas ce comportement 🤔 

>  Sur une dimension de performance rendering / painting.

Aujourd'hui l'ajout d'une div avec une class niveau perf est très très négligeable. Même si y'a 20 checkbox dans une page ça resterait négligeable ... (ex: https://stackoverflow.com/questions/6106219/does-the-use-of-multiple-divs-affect-page-performance) 

Après au niveau "joli" quand on code pour pas à avoir un sapin de Noël de div, je compte principalement sur le fait que dans nos cas d'utilisations de la div on n'ait pas à modifier quoi que ce soit 😅
Je suis totalement pour enlever les divs quand elles ne sont pas nécessaire mais là dans ce cas là je trouve que le `...attributes` sur l'input apporte bien plus en plus value que la gêne de devoir rajouter une div

Reply:
En fait si vraiment le style de la div parente pose soucis autant rajouter un attribut supplémentaire et optionnel à la checkbox pour filer une classe qui permet de custom la `<div class="pix-checkbox />` ?

Reply:
> En fait si vraiment le style de la div parente pose soucis autant rajouter un attribut supplémentaire et optionnel à la checkbox pour filer une classe qui permet de custom la `<div class="pix-checkbox />` ?

yep c'était dans les suggestions de départ. Après ça risque d'être compliqué puisque le composant est par défaut en flex. on aurait la seul possibilité de le mettre en `inline-flex` dans le cas ou on ne veut pas qu'il soit en ligne complète.

> Les seules checkbox que je connais sur Pix App sont celle dans le formulaire d'inscription et celles dans les épreuves et dans les deux cas on veut qu'elles prennent toute la largeur du parent j'ai l'impression.
Du coup j'ai du mal à voir dans quels autres cas on ne veut pas ce comportement 🤔

Ça reste que mon avis personnel aussi, je me dit que les composants que l'on crée devrait avoir le même comportement basique que leur homologue. ça laisse la possibilité de les rendre block si besoin, sans se dire. 
C'est encore une disparité que l'on retrouve sur nos composants certains sont block d'autres non, sur une même topologie d'élément.
On peut éventuellement faire un point pour statuer sur une pratique "uniforme" ?

Reply:
Vu ensemble : proposition d'un point pratique de code + rajout d'un attribut pour surcharger la classe parente en attendant d'opter pour une pratique